src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1286,14 +1297,37 @@ public Worker build() {
                 workerStateChangeListener = DEFAULT_WORKER_STATE_CHANGE_LISTENER;
             }
 
-            if(leaseCleanupValidator == null) {
-                leaseCleanupValidator = DEFAULT_LEASE_CLEANUP_VALIDATOR;
+            if ((shardSyncer == null && leaseCleanupValidator != null) || (shardSyncer != null && leaseCleanupValidator == null)) {
+                throw new IllegalArgumentException("Either both ShardSyncer and LeaseCleanupValidator should be injected, or neither of them.");
+            }
+            else {
+                if (shardSyncer == null) {
+                    leaseCleanupValidator = DEFAULT_LEASE_CLEANUP_VALIDATOR;
+                    shardSyncer = new KinesisShardSyncer(leaseCleanupValidator);
+                }
             }
 
             if(leaseSelector == null) {
                 leaseSelector = DEFAULT_LEASE_SELECTOR;
             }
 
+            if (leaseTaker == null) {
+                leaseTaker = new LeaseTaker<>(leaseManager, leaseSelector, config.getWorkerIdentifier(), config.getFailoverTimeMillis())
+                        .withMaxLeasesForWorker(config.getMaxLeasesForWorker())
+                        .withMaxLeasesToStealAtOneTime(config.getMaxLeasesToStealAtOneTime());
+            }
+
+            // We expect users to either inject both LeaseRenewer and the corresponding thread-pool, or neither of them (DEFAULT).
+            if ((leaseRenewerThreadPool == null && leaseRenewer != null) || (leaseRenewerThreadPool != null && leaseRenewer == null)) {
```

Comment:
I see this pattern of null checking is repeated more than once. Would rather make it reusable via a common method like verifyRequiredArgsAndInit. If the refactoring is not beneficial, ignore.

Reply:
Discussed offline. Leaving as-is for now.